### ISSUE TITLE
Resource dependencies

### ISSUE THREAD

**Issue by floriandejonckheere** [floriandejonckheere/kozo#52]
As a developer, I want to (implicitly) define dependencies between resources.

```ruby
resource "hcloud_ssh_key", "default" do |s|
  s.name = "default"
  s.public_key = File.read("id_rsa.pub")
end

resource "hcloud_server", "default" do |s|
  s.name = "default"
  s.ssh_keys = [hcloud_ssh_key.default]
end
```

Server depends on SSH key.

**Phase one**

Add "reference" option to attributes. Add appropriate methods.
Allow reference attributes to resolve at plan time:

```ruby
r.ssh_keys = [hcloud_ssh_key.default]
```

References are resolved in two places:
- When displaying resources => look up resource in unified state, apply attribute from resource (might change resulting attribute or not)
  E.g. `hcloud_ssh_key.default` resolves to "(known after apply)"

- When planning: plan dependency creation/modification first
  E.g. `hcloud_ssh_key.default` => creation/modification should be planned first, resolve attribute afterwards

**Phase two**

- Allow all attributes to resolve at plan time:
  `r.name = hcloud_ssh_key.default.name`

**Comment by floriandejonckheere** [comment]
Dependent resource destruction is still not supported.

**Comment by floriandejonckheere** [comment]
Closed in favour of #61.
